mat_vec.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <vector>
+
+std::vector<double> mat_vec(const std::vector<std::vector<double>>& mat, const std::vector<double>& vec) {
```

Comment:
The naming of this function is not ideal. If you did not have the context of the assignment, you would not know what this function does without reading the code. Consider renaming it to `matrix_vector_product`, or `inner_product`, the latter you would use the same name for matrix-vector and matrix-matrix product but take advantage of function overloading.